src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2829,7 +2829,8 @@ public bool HasImplicitTypeParameterConversion(TypeParameterSymbol source, TypeS
                 return true;
             }
 
-            if ((destination.TypeKind == TypeKind.TypeParameter) &&
+            if (destination is TypeParameterSymbol { AllowByRefLike: false } &&
+                !source.AllowByRefLike &&
```

Comment:
>!source.AllowByRefLike && [](http://example.com/codeflow?start=16&length=25)

Are we testing this case?

Reply:
> Are we testing this case?

I tried, but given the convoluted nature of the conversion classification algorithms, I was not able to find a scenario where we get here with ```source.AllowByRefLike``` equals `true`. Nevertheless, I am fairly comfortable making this change regardless of the fact.